src/core_scan/Core_scan.ml

```diff
@@ -852,6 +852,8 @@ let scan ?match_hook config ((valid_rules, invalid_rules), rules_parse_time) :
 
   let all_targets = targets @ new_extracted_targets in
 
+  let rule_filter_cache = Hashtbl.create (List.length valid_rules) in
```

Comment:
We need a changelog, at least say that we fixed a perf regression and specify in which version was it introduced.